core/vm/ewasm.go

```diff
@@ -145,3 +175,14 @@ func (in *InterpreterEWASM) CanRun(file []byte) bool {
 
 	return true
 }
+
+// PostContractCreation meters the contract once its init code has
+// been run.
+func (in *InterpreterEWASM) PostContractCreation(code []byte) []byte {
+	if in.metering {
+		// hera currently seems to ignore the error code
```

Comment:
Can you create an issue on hera for this please?

Reply:
apparently now the result is checked with `ensureCondition`; either it's been fixed since I wrote that or I missed it previously. The only check is, however, that [the code size is greater than 5 bytes](https://github.com/ewasm/hera/blob/f08ca8296626e7020915272453fca52fde9ccca8/src/hera.cpp#L321). I would need to know if the metering contract only returns 4 bytes upon error.

Reply:
The semantics are not properly defined. It checks for 5 bytes because that is the basic wasm header (magic + version). The code itself returns 0 bytes if there was an issue with the call, but assumes if the call to sentinel succeeds then its output is correct.

i.e. the sentinel contract should do a revert or abort in case failure

Reply:
Sorry for chiming in, but wasm header should be 8 bytes: `"\00asm\01\00\00\00"`

Reply:
It ignores the section header size, etc.

Reply:
It was just a quick measure to have some kind of error handling. Here's an issue to discuss the proper semantics: https://github.com/ewasm/design/issues/150

Reply:
> It ignores the section header size, etc.

Yeah, but just to be clear: both magic and version are `u32`.